kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -39,11 +44,30 @@ public class SimpleAclAuthorizer implements Authorizer {
     private final AuthorizationService authorizationService;
 
     private final boolean forceCheckGroupId;
+    // Cache the authorization results to avoid authorizing PRODUCE or FETCH requests each time.
+    // key is (topic, role)
+    private final LoadingCache<Pair<TopicName, String>, Boolean> produceCache;
+    // key is (topic, role, group)
+    private final LoadingCache<Triple<TopicName, String, String>, Boolean> fetchCache;
 
     public SimpleAclAuthorizer(PulsarService pulsarService, KafkaServiceConfiguration config) {
         this.pulsarService = pulsarService;
         this.authorizationService = pulsarService.getBrokerService().getAuthorizationService();
         this.forceCheckGroupId = config.isKafkaEnableAuthorizationForceGroupIdCheck();
+        this.produceCache = config.getAuthorizationCacheBuilder()
+                .build(new CacheLoader<>() {
+                    @Override
+                    public @Nullable Boolean load(Pair<TopicName, String> topicNameStringPair) {
+                        return null;
+                    }
+                });
+        this.fetchCache = config.getAuthorizationCacheBuilder()
+                .build(new CacheLoader<>() {
+                    @Override
+                    public @Nullable Boolean load(Triple<TopicName, String, String> topicNameStringStringTriple) {
+                        return null;
+                    }
+                });
```

Comment:
```suggestion
        this.produceCache = config.getAuthorizationCacheBuilder().build();
        this.fetchCache = config.getAuthorizationCacheBuilder().build();
```
We don't need to use a loading cache here.